requires.py

```diff
@@ -12,139 +12,183 @@
-            conv.remove_state('{relation_name}.registry_location.available')
+            clear_flag(
+                self.expand_name(
+                    '{endpoint_name}.connected'))
+            clear_flag(
+                self.expand_name(
+                    '{endpoint_name}.dns.available'))
+            clear_flag(
+                self.expand_name(
+                    '{endpoint_name}.auth.available'))
+            clear_flag(
+                self.expand_name(
+                    '{endpoint_name}.cluster_tag.available'))
+            clear_flag(
+                self.expand_name(
+                    '{endpoint_name}.registry_location.available'))
 
     def check_states(self):
-        """Toggle states based on available data.
-
         """
-        conv = self.conversation()
+        Toggle states based on available data.
+        """
         if self.dns_ready():
-            conv.set_state('{relation_name}.dns.available')
+            set_flag(
+                self.expand_name(
+                    '{endpoint_name}.dns.available'))
         else:
-            conv.remove_state('{relation_name}.dns.available')
+            clear_flag(
+                self.expand_name(
+                    '{endpoint_name}.dns.available'))
 
         if self._has_auth_credentials():
-            conv.set_state('{relation_name}.auth.available')
+            set_flag(
+                self.expand_name(
+                    '{endpoint_name}.auth.available'))
         else:
-            conv.remove_state('{relation_name}.auth.available')
+            clear_flag(
+                self.expand_name(
+                    '{endpoint_name}.auth.available'))
 
         if self.get_cluster_tag():
-            conv.set_state('{relation_name}.cluster_tag.available')
+            set_flag(
+                self.expand_name(
+                    '{endpoint_name}.cluster_tag.available'))
         else:
-            conv.remove_state('{relation_name}.cluster_tag.available')
+            clear_flag(
+                self.expand_name(
+                    '{endpoint_name}.cluster_tag.available'))
 
         if self.get_registry_location():
-            conv.set_state('{relation_name}.registry_location.available')
+            set_flag(
+                self.expand_name(
+                    '{endpoint_name}.registry_location.available'))
         else:
-            conv.remove_state('{relation_name}.registry_location.available')
+            clear_flag(
+                self.expand_name(
+                    '{endpoint_name}.registry_location.available'))
 
     def get_auth_credentials(self, user):
-        """ Return the authentication credentials.
-
         """
-        conv = self.conversation()
-        remote_creds = conv.get_remote('creds')
-        if not remote_creds:
+        Return the authentication credentials.
+        """
+        rx = self.all_joined_units.received_raw.get('creds')
```

Comment:
I made a mistake earlier when I said this should be `received_raw`.  This is the one case where JSON encoding was being used, so it should just be `self.all_joined_units.received.get('creds')`.  Sorry.